presto-hive/src/test/java/com/facebook/presto/hive/unittests/TestHiveMetadata.java

```diff
@@ -0,0 +1,143 @@
+import com.google.common.collect.ImmutableMap;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.List;
+import java.util.Map;
+
+import static com.facebook.presto.connector.meta.ConnectorFeature.ADD_COLUMN;
+import static com.facebook.presto.connector.meta.ConnectorFeature.CREATE_SCHEMA;
+import static com.facebook.presto.connector.meta.ConnectorFeature.CREATE_TABLE;
+import static com.facebook.presto.connector.meta.ConnectorFeature.CREATE_TABLE_AS;
+import static com.facebook.presto.connector.meta.ConnectorFeature.DROP_SCHEMA;
+import static com.facebook.presto.connector.meta.ConnectorFeature.DROP_TABLE;
+import static com.facebook.presto.connector.meta.ConnectorFeature.READ_DATA;
+import static com.facebook.presto.connector.meta.ConnectorFeature.RENAME_COLUMN;
+import static com.facebook.presto.connector.meta.ConnectorFeature.RENAME_SCHEMA;
+import static com.facebook.presto.connector.meta.ConnectorFeature.RENAME_TABLE;
+import static com.facebook.presto.connector.meta.ConnectorFeature.WRITE_DATA;
+import static com.facebook.presto.hive.HiveQueryRunner.TIME_ZONE;
+import static com.facebook.presto.hive.HiveStorageFormat.TEXTFILE;
+import static com.facebook.presto.hive.HiveTableProperties.BUCKETED_BY_PROPERTY;
+import static com.facebook.presto.hive.HiveTableProperties.BUCKET_COUNT_PROPERTY;
+import static com.facebook.presto.hive.HiveTableProperties.PARTITIONED_BY_PROPERTY;
+import static com.facebook.presto.hive.HiveTableProperties.STORAGE_FORMAT_PROPERTY;
+import static com.facebook.presto.spi.type.VarcharType.VARCHAR;
+import static com.google.common.collect.Iterables.getOnlyElement;
+
+@SupportedFeatures({
+        READ_DATA,
+        WRITE_DATA,
+        CREATE_SCHEMA,
+        DROP_SCHEMA,
+        RENAME_SCHEMA,
+        CREATE_TABLE,
+        CREATE_TABLE_AS,
+        DROP_TABLE,
+        RENAME_TABLE,
+        ADD_COLUMN,
+        RENAME_COLUMN})
+public class TestHiveMetadata
+        implements BaseMetadataTest, MetadataSchemaTest, MetadataTableTest
+{
+    private Connector connector;
+
+    @BeforeAll
+    public void beforeAll()
+            throws Exception
+    {
+        this.connector = createHiveConnector();
+    }
+
+    @AfterAll
+    public void cleanUp()
+            throws Exception
+    {
+        connector.shutdown();
+    }
+
+    @Override
+    public Connector getConnector()
+    {
+        return connector;
+    }
+
+    @Override
+    public Map<String, Object> getTableProperties()
+    {
+        return ImmutableMap.of(
+                BUCKETED_BY_PROPERTY, ImmutableList.of(),
+                BUCKET_COUNT_PROPERTY, 0,
+                PARTITIONED_BY_PROPERTY, ImmutableList.of(),
+                STORAGE_FORMAT_PROPERTY, TEXTFILE);
+    }
+
+    @Override
+    public List<ColumnMetadata> withSystemColumns(List<ColumnMetadata> connectorColumns)
```

Comment:
> Or just add getConnectorSpecificColumns method and to the concatenation of lists in tests?

That's how it was originally written. I changed it because this covers cases where the modification isn't just tacking some more columns on at the end of the list.

I've changed the name.